transport/src/main/java/io/netty/channel/ChannelHandlerInvokerUtil.java

```diff
@@ -27,144 +27,227 @@
  */
 public final class ChannelHandlerInvokerUtil {
 
+    /**
+     * Makes best possible effort to detect if {@link ChannelHandler#handlerAdded(ChannelHandlerContext)} was called
+     * yet. If not return {@code false} and if called or could not detect return {@code true}.
+     *
+     * If this method returns {@code true} we will not invoke the {@link ChannelHandler} but just forward the event.
+     * This is needed as {@link DefaultChannelPipeline} may already put the {@link ChannelHandler} in the linked-list
+     * but not called {@link }
+     */
+    private static boolean isAdded(ChannelHandlerContext ctx) {
```

Comment:
@trustin I was thinking about if this should life in `DefaultChannelHandlerInvoker` or here and decide to put it better here so it will work the same way if the user will implement his own `ChannelHandlerInvoker` and uses this util.